casr/src/bin/casr-cluster.rs

```diff
@@ -63,37 +63,28 @@ fn make_clusters(
         bail!("{} reports, nothing to cluster...", len);
     }
 
-    casreps.sort_by(|a, b| {
-        a.file_name()
-            .unwrap()
-            .to_str()
-            .unwrap()
-            .cmp(b.file_name().unwrap().to_str().unwrap())
-    });
-
     // Start thread pool.
     let custom_pool = rayon::ThreadPoolBuilder::new()
         .num_threads(jobs.min(len))
         .build()
         .unwrap();
 
-    // Stacktraces from casreps
-    let traces: RwLock<Vec<Stacktrace>> = RwLock::new(Vec::new());
-    // Crashlines from casreps
-    let crashlines: RwLock<Vec<String>> = RwLock::new(Vec::new());
-    // Casreps with stacktraces, that we can parse
-    let filtered_casreps: RwLock<Vec<PathBuf>> = RwLock::new(Vec::new());
+    // Report info from casreps: (casrep, (trace, crashline))
+    let mut casrep_info: RwLock<Vec<(PathBuf, (Stacktrace, String))>> = RwLock::new(Vec::new());
     // Casreps with stacktraces, that we cannot parse
     let mut badreports: RwLock<Vec<PathBuf>> = RwLock::new(Vec::new());
     custom_pool.install(|| {
         (0..len).into_par_iter().for_each(|i| {
             if let Ok(report) = util::report_from_file(casreps[i].as_path()) {
                 if let Ok(trace) = report.filtered_stacktrace() {
-                    traces.write().unwrap().push(trace);
-                    filtered_casreps.write().unwrap().push(casreps[i].clone());
+                    let mut crashline = String::new();
                     if dedup {
-                        crashlines.write().unwrap().push(report.crashline);
+                        crashline = report.crashline;
```

Comment:
we may always save crash line, that's not much time and memory consuming

Reply:
Done